.github/workflows/pre-merge.yaml

```
@@ -0,0 +1,51 @@
+name: Pre Merge Checks
+on:
+  pull_request:
+    branches:
+      - '*'
```

Does the branch wildcard doe anything special?

Additionally I'm thinking it might make sense to also include `push`

```suggestion
on: [pull_request, push]
```

---

I generally prefer to restrict `push` to `master` only. Otherwise you'll have Pull Requests with duplicated Status reports (one for the `pull_request` trigger and another for the `push` trigger)